spark-3.1-spanner-lib/src/test/java/com/google/cloud/spark/spanner/ReadIntegrationTestBase.java

```diff
@@ -58,39 +61,232 @@ public void testReadFromCompositeTable() {
+    gotIs.addAll(row2.getList(0));
+    Collections.sort(gotIs);
+
+    List<Timestamp> wantIs = new ArrayList();
+    Arrays.asList(
+            ZonedDateTime.parse("2023-08-26T12:11:10Z"),
+            ZonedDateTime.parse("2023-08-27T12:11:09Z"),
+            ZonedDateTime.parse("2023-09-22T12:11:10Z"),
+            ZonedDateTime.parse("2023-09-23T12:11:09Z"))
+        .forEach(zd -> wantIs.add(new Timestamp(zd.toInstant().toEpochMilli())));
+    Collections.sort(wantIs);
+    assertThat(gotIs).containsExactlyElementsIn(wantIs);
+  }
+
+  @Test
+  public void testValueLimits() {
+    Dataset<Row> df = readFromTable("valueLimitsTable");
+    assertThat(df.count()).isEqualTo(4);
+
+    // 1. Test the limits of INT64.
+    List<Long> gotAs = df.select("A").as(Encoders.LONG()).collectAsList();
+    Collections.sort(gotAs);
+
+    long[] lA = {Long.MIN_VALUE, Long.MAX_VALUE, 0, 1};
+    List<Long> wantAs = Arrays.stream(lA).boxed().collect(Collectors.toList());
+    Collections.sort(wantAs);
+    assertThat(wantAs).containsExactlyElementsIn(gotAs);
+
+    // 2. Test the limits of FLOAT64.
+    List<Double> gotBs = df.select("B").as(Encoders.DOUBLE()).collectAsList();
+    Collections.sort(gotBs);
+
+    List<Double> wantBs =
+        Arrays.asList(Double.NaN, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY, 0.657818);
+    Collections.sort(wantBs);
+    assertThat(wantBs).containsExactlyElementsIn(gotBs);
+
+    // 3. Test the limits of NUMERIC.
+    List<BigDecimal> gotCs = df.select("C").as(Encoders.DECIMAL()).collectAsList();
+    Collections.sort(gotCs);
+
+    List<BigDecimal> wantCs =
+        Arrays.asList(
+            asSparkBigDecimal("-9.9999999999999999999999999999999999999E+28"),
+            asSparkBigDecimal("9.9999999999999999999999999999999999999E+28"),
+            asSparkBigDecimal("10.389"),
+            asSparkBigDecimal("-10.389"));
+    Collections.sort(wantCs);
+    assertThat(wantCs).containsExactlyElementsIn(gotCs);
+  }
+
+  @Test
+  public void testBytes() {
+    Dataset<Row> df = readFromTable("bytesTable");
+    assertThat(df.count()).isEqualTo(3);
+
+    List<Row> allAs = df.select("A").collectAsList();
+    assertEquals(allAs.size(), 3);
+    GenericArrayData a0 = allAs.get(0).getAs(0);
+    assertThat(a0.getBinary(0)).isEqualTo("ABCDEFGHIJKLMNOPQ".getBytes());
+    GenericArrayData a1 = allAs.get(1).getAs(0);
+    assertThat(a1.getBinary(0)).isEqualTo("abcdefghijklmnopq".getBytes());
+    GenericArrayData a2 = allAs.get(2).getAs(0);
+    assertThat(a2.getBinary(0)).isEqualTo("1234efghijklmnopq".getBytes());
+  }
+
+  @Test
+  public void testDateLimits() {
+    Dataset<Row> df = readFromTable("valueLimitsTable");
+    assertThat(df.count()).isEqualTo(4);
+
+    // 1. Test the limits of INT64.
+    List<Date> gotDs = df.select("D").as(Encoders.DATE()).collectAsList();
     Collections.sort(gotDs);
+
+    List<Date> expectDs =
+        Arrays.asList(
+            Date.valueOf("1699-12-31"),
```

Comment:
Does Spark spanner connector support 0001-01-01 and 9999-12-31 ? If it doesn't, we should call it out in the public doc.